WeakAuras/GenericTrigger.lua

```diff
@@ -2527,7 +2527,16 @@ do
     end
   end
 
+  local lastCheckCooldownReady
   function Private.CheckCooldownReady()
+    -- skip if function was called on same frame
+    local now = GetTime()
```

Comment:
If there are multiple events in the same frame, we want to handle the "last" one, as presumably the first event is fired before everything is updated.

Reply:
I don't believe that is an issue

using this macro
```
/use 13
/use 14
/cast Ancestral Guidance
/cast Astral Shift
/cast Ascendance
/cast Spiritwalker's Grace
```
and this aura
```
!WA:2!LF5wVTXXvC8SG11URtvLOSvTDCtyTAuSrBDuKRnC9BKYuYYHMuEjLTC6fXH7oK7yTC31ZmlPP6vXQ6QEVv9ERBlclssVeGwi0xZl6tWbef5dGEOVga)wERNz2Lsw2rTiafOfWVio7mZz2ZLFN)ZkJBKwm1LPTs3mTtAN1FXniC7cu)gs31gB7AbChkphXEzhEqy4W58yRScH7KPsqGNKfYVBP61fuPbVtYGnj(2Ub85dy(YAtNVyL8wBgFkth4fWVQHHrklriXMEaoXwYc8fxWsijCPzoMptAwRo(JW1SRKZA0GYfNycEYW71ZHwlQE9kDcP8RKVW8ZSqHCs8blseNmvxri1ZBohH52IOA0wuFzzC3S7U5stNTCLLkxjRvLCr4lPwOhPdLVZ2MNtXTzvE(8fkyPNP2vOepPRLpPjvyUrKFIlyUPDGxut)YQii1dWGKY9jE3a9tmuwDdIpRjrfvxaE6zGrZ5h4t3Ise0YsUoPEW(oOVQ2XskpNlO2b(oIUQ9OIly8CnjmF0wyC4tCq45HjGxaho0JoJfXJ1WhoehtEkpZOhgnw0gQmQzoVaIZm5eSvOJz1mclvM5WSJ9yWznTS9icHAunjXdJv1WUn4mh17)uwxUFTK6DCflvQuZNAYZEEECSrDTe2ODP6ztvrV2Ocw1IlYU9I)nVtdA4iLVteHtZmtKNxMB6YK0ECTdQSzlNoyYLz3GhefMyvz0FpqoraxcJ3nsqlWAYKUD5bsDgZOgN4WIeR909QJh50mUThT62yceZb(YzqEkGxZk7LNBHYwEkBp8wXhCmC(HsiX58Xhs1tgy3kUUDmJ79ANI5STKkKzSr02jOTVydvo5MmhP7H3SohbbSgsKKupigXNrnLksQvEAR85lIUz7ySyQiMZMLMBQsx8c36g86o1qpSoRHzFsKenSuOmUiHHE7ExPK1CVsPIvYwONIeyXRTjZVEapgLm3qq9QR7OGdL2(2RpEpgU1YbrCB6jFk46MWh0aoiCO(dC9YTzH0QBn4rvXWfomIKMWh2egYe(ifNagE90DXssZIyyyeocMUxQb1NYz2LDdAxYVA)98ypb(twV2Koc4eWZOnTSxG8z2q32PoLJzC(6iZRwbgd(OWjdp10jUqM55bn4uHiZPZFNiwytKDYOm)mWiq6(CSBR8GJj8iLdzCMSnXBzk)fezMfvkO9vo4oVQQWrn3snZGiOkCm44WZITQ4KyDYE5Q7SU69ufEoZnDyIBh5JQoTOtr87eocrpUsCV91cCOV5tbF8vFWYuAywv)I0svbCHtf3oV3wYt)FUjffSeePQHhlaFYlngmAFSlv1FPWEFN1psyAj9UYLeUemnTyIiA4X1tQ)tmgSKizam(g7S2PEEr4WpK57QWUD8SdWMCzxOsPWr1tImyGIRS1KDUS4Jpi(fXUlng3xTFmHFRe3zl963osiz17ahQNUFKwbNmxXsfZdJYXGsTNE6nApq2i1gjrGVmC4z4SvYC9iIJQhktLk7jWh8M6RNRn2LEtojS7ntge7aTycwnSHpjMFOlAchzUIfZBTuUsvQu6AwZn7vQ0FNxTsu54jr0I759OwvFxsPfQuyUI5F7(n8cApdNENiQVDN5)atE2PoFC0PwqXtZ6LKMD7QMszD)ArszGFjulbVvPNA647pp6MQXdcTnAmWuDQrV2a3jN6bx9okR1w1osfxM9Y(yptk9kfy44r0NFUyT2AUuwdx5QWNQvpn2RLZD3wlWJ5hhv)I9e1em)gE01gg15Hxcv7ht1OwfMuj8dVO6VHPv2Vx7QcFA9gUuoLAdmVjC)QDhiPu1QTMuYTsqqtd4gWHGztur5cSDlZlDoRD5a9XpqkaBZWUkSH9LHc4LyHUKuqXnV(kL6Wd7CTtxCfoZxPlJ74QPGdyavoXew2Uu7LVGvlIxenv3weoJGOqVa)bNkmCkUTlXVbf)Gb4nUQ(kRD)tQUH4fAuUSd8QWcMHhzqGOO4lZeQtZXD9XHBIcQXQPWIvHBnq2CiTYzISj8kgWNRk85HVGsje(IFmLgg25xnw1diqJQqnWgCuYuagipRPwEct6lPfjFotWfyWT1YnWYdeyqHdL(XdjW8iZm0JnJxIUc0C9Ja(gqaoxikma3rleaCqasicATk02aUlgwDGvWT)LGVSU0aFf4RcFnd4RdRcDRcFdyn4BEC4EgW3cw)THV9GEa47a)yd47bFF4hCu4hAa)OyJ)UUWgUWpnf8tsb)SrGFUl8lu0yc58RMa(1pg19B2d4b)sx4sW9ra4ClCTBLBXPYEMl)zH30fRrAOcjGFVbcEpkD9zu361Rk87wf(dPuS5RP5j4vJ9lezGx)cWBKc(JWFcrd4p)EcfWFbPHFBveYsQ8jMV)aWqia8pHrT0xePl3QspcdNm84Vh31PVt7mAkPwy6S(24hyWjEzMf)cburIMqik2qrtiMeJiV(7QqKN0WJx)DRMqf7aj7GhExS8KxCYRhj7SOgmq(ajaKnqwzaBeEKhpdNzkLecIrk6yhubXgeEEFsj3V67tkb(R7HqG)wmiSv2yxSSlRUCFjG35jsc4D2FcOWlFTw3Xzf3l2(C7lba)9)VRA3lRa)hL(32T)wR9KyT(TwB)R1NRo(DfDOZwxS8(wR3A3e7)tAYlUNYE896jfEvfE37)X68GpvaR3k(4j4R)3VkEZtF18tkMPUtl2(3DB)FXYC6w)Jf)x
```
everything goes on cooldown at once as expected